src/api.v2/helpers/worker/workSubmit/createWorkerK8s.js

```diff
@@ -62,7 +104,14 @@ const createWorkerResources = async (service) => {
     return { name, creationTimestamp, phase };
   }
 
-  const pods = await getAvailablePods(namespace);
+  // scale pods if worker has zero replicas
+  let pods = await getAvailablePods(namespace);
+  const { spec: { replicas } } = await getDeployment('worker', namespace);
```

Comment:
yes it would although so would e.g. `getPods` (called by `getAssignedPods` and `getAvailablePods`) . Not clear to me why we need be especially cautions about coding for an unhandled error in this case but not the other. If there is no deployment, there will also be no pods and so the failure would likely happen upstream no?

Reply:
It is not like we need to be especially cautious for this case -- it is in general a best practice to handle possible error cases in the code.  We should have been cautious for all cases. If there is no error handling for some cases, where the code can fail, this it is something that needs to be corrected, not used as an example.

Reply:
Of course, there are some limits to everything, but in my opinion in the api we definitely don't have error overhandling problem. In general, I don't think I've seen a piece of software that suffers from too much error handling :D

Reply:
If there is no deployment, wouldn't you want to raise an Exception here?

Reply:
I am thinking from the perspective of the other environments (all non HMS productions) that don't require this scaling -- is there a possibility that `getDeployment` returns nothing, or returns an exception, but still we would want to proceed with the rest of the code?

There is handling, see line 115-117 in this file for example:
```
  if (pods.length < 1) {
    throw new Error(`Experiment ${experimentId} cannot be launched as there are no available workers.`);
  }
```

Reply:
I added error handling to ensure that `getDeployment` can fail and the function will still proceed